Remove GCP code from repo and use GCP stack going forward

jbw976 opened the pull request:
### Description of your changes

This PR removes all GCP code from the repo and adds instructions for how to use the new GCP stack.  The [stack-gcp repo](https://github.com/crossplaneio/stack-gcp/) now has an official CI build on Jenkins, so new versions of the stack are being automatically built and published.

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation], [examples], or [release notes].
- [x] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplaneio/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

displague commented on docs/install-crossplane.md:
```diff
@@ -49,6 +49,45 @@ For example:
+
+#### Uninstall
+
+The GCP stack can be uninstalled simply by deleting the stack resources from the cluster:
```
Another sentence about what will happen to provisioned resource could be helpful here.

hasheddan commented on pkg/controller/aws/aws.go:
```diff
@@ -28,7 +28,7 @@ import (
 type Controllers struct{}
 
-// SetupWithManager adds all GCP controllers to the manager.
+// SetupWithManager adds all AWS controllers to the manager.
```
Thank you for catching this here and in other the other controllers!

displague commented on docs/install-crossplane.md:
```diff
@@ -49,6 +49,45 @@ For example:
 
+## Installing Cloud Provider Stacks
+
+After Crossplane has been installed, you can add additional functionality to its control plane by installing Crossplane Stacks.
```
Somewhere in the example instructions we should link back to this heading..
"Install Crossplane (some existing link) **and  the GCP Cloud provider stack (link to this section)** ... "

I suppose it would be more hindrance than helpful if we included a StackRequest/StackInstall record in the example provider.yaml files, unless the StackRequest objects throughout the examples were identically named in the same namespace.

jbw976 commented on docs/gitlab/gitlab-gcp.md:
```diff
@@ -30,10 +30,12 @@ kubectl config set-context [your-cluster-context]
+- Install the GCP stack into Crossplane using the [GCP stack section](../install-crossplane.md#gcp-stack) of the install guide.
+- Obtain [Cloud Provider Credentials](../cloud-providers.md)
 
-#### GCP Provider   
```
some whitespace only changes that were introduced by my auto editor settings.  The main change to this file was to add the link to the GCP stack install section just above.

jbw976 commented:
@displague I have incorporated all of your feedback in a new commit.  I clarified that uninstall can delete provisioned resources from GCP, I linked in a few places to this GCP stack install doc section, and I added an example stack-gcp.yaml to the examples.  Please take another look!
